src/Uno.UI/UI/Xaml/Shapes/Shape.Android.cs

```diff
@@ -26,23 +26,26 @@ protected Windows.Foundation.Rect GetDrawArea(Android.Graphics.Canvas canvas)
 		{
 			var drawSize = default(Windows.Foundation.Size);
 
+			var suggestedWidth = double.IsNaN(Width) ? canvas.Width : Width;
```

Comment:
/cc @carldebilly @dr1rrb

Reply:
This change was due to the fact that when rendered by explicit request with full size canvas, the rectangle tnen filled the given area even when it's width and height were smaller. Because of that the rectangle appeared to be much larger below the blur than it's real size

Reply:
#3693 this issue remains even after this fix, I didn't get to it yet

Reply:
Hm, had to revert the change, as some rectangles didn't render properly. For now, we can probably merge this with this limitation and improve later.

Reply:
We have already refactored the shapes measure/arrange for iOS and macOS. The code is now shared with Skia, we should complete the work and share it also for Android and WASM. So yeah I would recommend you to let this bug open if not too annoying.

Reply:
It is problematic when a shape is below acrylic brush, as then its size usually does not reflect reality, but I would say that is an acceptable limitation currently